burn-train/src/metric/mod.rs

```diff
@@ -6,20 +6,30 @@ pub mod state;
 
 mod acc;
 mod base;
+#[cfg(feature = "cli")]
 mod cpu_temp;
+#[cfg(feature = "cli")]
 mod cpu_use;
+#[cfg(feature = "cli")]
 mod cuda;
+#[cfg(feature = "cli")]
 mod gpu_temp;
 mod learning_rate;
 mod loss;
+#[cfg(feature = "cli")]
 mod memory_use;
 
 pub use acc::*;
 pub use base::*;
+#[cfg(feature = "cli")]
 pub use cpu_temp::*;
+#[cfg(feature = "cli")]
 pub use cpu_use::*;
+#[cfg(feature = "cli")]
 pub use cuda::*;
+#[cfg(feature = "cli")]
 pub use gpu_temp::*;
 pub use learning_rate::*;
 pub use loss::*;
+#[cfg(feature = "cli")]
 pub use memory_use::*;
```

Comment:
Except for the `cuda` metric, they should be enabled even without the cli feature.